themes/10up-theme/includes/blocks/tui-button/edit.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { RichText, useBlockProps } from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
+
+/**
+ * Edit component.
+ * See https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-edit-save/#edit
+ *
+ * @param {object}   props                  The block props.
+ * @param {object}   props.attributes       Block attributes.
+ * @param {string}   props.attributes.title Custom title to be displayed.
+ * @param {string}   props.className        Class name for the block.
+ * @param {Function} props.setAttributes    Sets the value for block attributes.
+ * @returns {Function} Render the edit screen
+ */
+const Edit = (props) => {
+	const { attributes, setAttributes } = props;
+	const { label } = attributes;
+
+	const blockProps = useBlockProps();
+
+	return (
+		<div {...blockProps}>
+			<RichText
+				tagName="h2"
+				placeholder={__('Learn more')}
+				value={label}
+				onChange={(label) => setAttributes({ label })}
+			/>
+			<ServerSideRender
```

Comment:
I don't think that we should use this approach here. The editorial experience should be inline. If that means that for the beginning we need to maintain both the frontend and editor js in the scaffold UI plugin I think that is still worth it because of the better editorial experience.

Reply:
Totally agree! I haven't worked on the editing experience. I've mostly done a test to see that the partial can be used the same way as we would in a template file. Editing experience needs to be cohesive. THANKS!